### ISSUE TITLE
Autocomplete for metrics in PromQL query when adding a dashboard

### ISSUE THREAD

**Issue by polyglothacker** [SigNoz/signoz#687]
## Is your feature request related to a problem? No

Prometheus and Grafana autocompletes the metric exported by Prometheus when we start typing.  This is a very useful feature as many metrics start with the same prefix.

## Describe the solution you'd like

Provide autocomplete drop-down like Grafana does (See image)

**Comment by polyglothacker** [comment]
![graf](https://user-images.githubusercontent.com/97145793/153024466-7a018048-4ef2-4647-840a-e9ee630b811f.jpg)

**Comment by pranay01** [comment]
Yeah, this is something which we have in our pipeline. Understand that finding and typing exact text may be too cumbersome for users

**Comment by SoniaisMad** [comment]
@pranay01 Can i try to work on this one ?

**Comment by pranay01** [comment]
@SoniaisMad Do you have prior understanding of Prometheus query language and Clickhouse - might be important for working on this issue.

**Comment by SoniaisMad** [comment]
@pranay01 I understand, i'll pass on this one, and take a much simple issue as i'm starting to contribute and don't know really well all the repository

**Comment by pranay01** [comment]
cool - there are some issues labeled as  good first issues which can be easier to get started with @SoniaisMad - https://github.com/SigNoz/signoz/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22

**Comment by makeavish** [comment]
Prometheus Integrates this using codemirror-promql:
https://github.com/prometheus/codemirror-promql
Maybe we can integrate this^

Prometheus has integrated it over frontend, eg: https://github.com/prometheus/prometheus/blob/431ea75a11ca165dad9dd5d629b3cf975f4c186b/web/ui/react-app/src/pages/graph/CMExpressionInput.tsx
